streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -495,9 +498,23 @@ private void replaceStreamThread(final Throwable throwable) {
         }
     }
 
+    private StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse getActionForThrowable(final Throwable throwable,
+                                                                                        final StreamsUncaughtExceptionHandler streamsUncaughtExceptionHandler) {
+        final StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse action;
+        // Exception might we wrapped within a StreamsException one
+        if (throwable.getCause() != null && EXCEPTIONS_NOT_TO_BE_HANDLED_BY_USERS.contains(throwable.getCause().getClass())) {
```

Comment:
Could you please put this check in a method with a meaningful name like `wrappedExceptionIsIn()`? Then the check would be 
```
if (wrappedExceptionIsIn(EXCEPTIONS_NOT_TO_BE_HANDLED_BY_USERS))
``` 
and you can remove all other inline comments because the code per se contains all needed info.